### DISCUSSION TITLE
Check jwt expired

### DISCUSSION THREAD

**Discussion by centratelemedia** [kjur/jsrsasign#571] [General]
i have jwt 
```
const token="eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhZG1pbjEiLCJhdWQiOiJBdXRoZW50aWNhdGlvbiIsImlhdCI6MTY2OTg4MjI4MywiZXhwIjoxNjY5ODg1ODgzLCJpZCI6MSwic2Vzc2lvbmlkIjoiNDI3MDQ0OTQtYzUwYS00YTA5LTg0ZjItMmQzYzc0OGU3OWJjIn0.ctjwuFBlya7nfLRmRNzFPjl6WH3sU-u4HNuN7JuMfHo"
```
validate by java component (io.jsonwebtoken:jjwt-impl:0.11.5), result is
```
JWT expired at 2022-12-01T09:11:23Z. Current time: 2022-12-01T21:58:14Z, a difference of 46011795 milliseconds.
```
echek by jsrsasign
```
 var isValid = KJUR.jws.JWS.verify(token, "U+6ooI7VFkMwUIq/M0fvM18OAkdX5VvUi2gVbLNyRoE=", ["HS256"]);
 result is true
```

is this bug or my mistake?

**Comment by kjur** [comment]
JWS.verify only checks a signature of JWS. As for JWT validation including time check, please use JWS.verifyJWT:
https://kjur.github.io/jsrsasign/api/symbols/KJUR.jws.JWS.html#.verifyJWT

Optional parameter "gracePeriod" and "verifyAt" can be used.
